Automattic/wp-calypso
# Settings: Change wording of blog_public checkbox

fixes #42180

#### Changes proposed in this Pull Request

Url: `/settings/general/example.wordpress.com`

**Old wording:** "Do not allow search engines to index my site"

**New wording:** "Discourage search engines from indexing my site"
with explanation: "This option does not block access to your site — it is up to search engines to honor your request."

---

**Detail:** 

This brings the wording in line with wp-admin from wordpress.org:
[code link](https://github.com/WordPress/WordPress/blob/53c909a8256bbd481656ae864a4b7eb5a22c3692/wp-admin/options-reading.php#L184).

The point of clarification being made is that even though we modify
robots.txt entries to disallow everything, and add a meta robots tag
with "noindex, nofollow", it can still be ignored by search engines.
Putting a sign on your open lawn that says "Do not tresspass" does not
actually stop people from walking forward.

From my research, the wayback machine on archive.org either ignores
robots.txt now or plans to in the future. [1] To be excluded,
archive.org wants the author to send them an individual email, which
they will "review". [2]

1: https://blog.archive.org/2017/04/17/robots-txt-meant-for-search-engines-dont-work-well-for-web-archives/
2: https://help.archive.org/hc/en-us/articles/360004651732-Using-The-Wayback-Machine

This is stemming from a user who had ticked the box, but still found
their site on archive.org, when they thought it would be blocked.  See
issue #42180.

For the unit tests, I needed to add {exact: false} to a few
getByLabelText() calls because the label now spans html elements because
of the `<FormSettingExplanation>` added.  It does not match text across
elements. [3]

3: https://testing-library.com/docs/dom-testing-library/api-queries

---

#### Testing instructions

* Visit calypso.
* Visit Manage -> Settings on the left bar.
* Scroll down to 'privacy' to see the new checkbox label.

![2020-07-29_09-37](https://user-images.githubusercontent.com/937354/88813979-2a5cd480-d17f-11ea-9582-a00c6d640bff.png)

#### Other points

This does add new translated strings.  I'm not sure how to handle that, or what needs to be done.
